File: library/src/main/java/com/pengrad/telegrambot/request/SendSticker.java
package com.pengrad.telegrambot.request;

import com.pengrad.telegrambot.impl.TelegramBotClient;

import java.io.File;

/**
 * stas
 * 5/2/16.
 */
public class SendSticker extends AbstractMultipartRequest<SendSticker> {

    public SendSticker(TelegramBotClient api, Object chatId, String sticker) {
        super(api, chatId, sticker);
    }

    public SendSticker(TelegramBotClient api, Object chatId, File sticker) {
        super(api, chatId, sticker);
    }

    public SendSticker(TelegramBotClient api, Object chatId, byte[] sticker) {
        super(api, chatId, sticker);
    }


    /**
     * 
     * @param emoji Emoji associated with the sticker; only for just uploaded stickers
     * @return
     */
    public SendSticker emoji(String emoji) {
        add("emoji", emoji);
        return this;
    }

    public SendSticker thumbnail(byte[] thumbnail) {
        return super.thumbnail(thumbnail);
    }

    public SendSticker thumbnail(File thumbnail) {
        return super.thumbnail(thumbnail);
    }

    @Override
    protected String getFileParamName() {
        return "sticker";
    }

    @Override
    public String getDefaultFileName() {
        return ContentTypes.PHOTO_FILE_NAME;
    }

    @Override
    public String getDefaultContentType() {
        return ContentTypes.PHOTO_MIME_TYPE;
    }

}

File: library/src/main/java/com/pengrad/telegrambot/request/SendSticker.kt
package com.pengrad.telegrambot.request

import com.pengrad.telegrambot.utility.kotlin.optionalRequestParameter
import java.io.File

class SendSticker private constructor(
    chatId: Long? = null,
    channelUsername: String? = null,

    stickerUrl: String? = null,
    stickerFile: File? = null,
    stickerBytes: ByteArray? = null
) : AbstractThumbnailedMultipartRequest<SendSticker>(
    chatId = chatId,
    channelUsername = channelUsername,

    contentParameterName = "sticker",
    contentUrl = stickerUrl,
    contentFile = stickerFile,
    contentBytes = stickerBytes,

    defaultFileName = ContentTypes.PHOTO_FILE_NAME,
    defaultContentType = ContentTypes.PHOTO_MIME_TYPE
) {

    constructor(chatId: Long, stickerUrl: String) : this(
        chatId = chatId,
        channelUsername = null,
        stickerUrl = stickerUrl
    )

    constructor(channelUsername: String, stickerUrl: String) : this(
        chatId = null,
        channelUsername = channelUsername,
        stickerUrl = stickerUrl
    )


    constructor(chatId: Long, stickerFile: File) : this(
        chatId = chatId,
        channelUsername = null,
        stickerFile = stickerFile
    )

    constructor(channelUsername: String, stickerFile: File) : this(
        chatId = null,
        channelUsername = channelUsername,
        stickerFile = stickerFile
    )


    constructor(chatId: Long, stickerBytes: ByteArray) : this(
        chatId = chatId,
        channelUsername = null,
        stickerBytes = stickerBytes
    )

    constructor(channelUsername: String, stickerBytes: ByteArray) : this(
        chatId = null,
        channelUsername = channelUsername,
        stickerBytes = stickerBytes
    )


    @Deprecated("Use constructor with chatId or channelUsername instead", ReplaceWith("SendSticker(chatId, sticker)"))
    constructor(chatId: Any, sticker: String) : this(
        chatId = (chatId as? Number)?.toLong(),
        channelUsername = chatId as? String,
        stickerUrl = sticker
    ) {
        checkDeprecatedConstructorParameters()
    }

    @Deprecated("Use constructor with chatId or channelUsername instead", ReplaceWith("SendSticker(chatId, sticker)"))
    constructor(chatId: Any, sticker: File) : this(
        chatId = (chatId as? Number)?.toLong(),
        channelUsername = chatId as? String,
        stickerFile = sticker
    ) {
        checkDeprecatedConstructorParameters()
    }

    @Deprecated("Use constructor with chatId or channelUsername instead", ReplaceWith("SendSticker(chatId, sticker)"))
    constructor(chatId: Any, sticker: ByteArray) : this(
        chatId = (chatId as? Number)?.toLong(),
        channelUsername = chatId as? String,
        stickerBytes = sticker
    ) {
        checkDeprecatedConstructorParameters()
    }

    @Suppress("NOTHING_TO_INLINE")
    private inline fun checkDeprecatedConstructorParameters() {
        if (this.chatId == null && this.channelUsername == null) {
            throw IllegalArgumentException("chatId parameter must be either Long or String")
        }
    }

    var emoji: String? by optionalRequestParameter()

    fun emoji(emoji: String) = applySelf { this.emoji = emoji }

}